libraries/xcode.rb

```diff
@@ -6,12 +6,18 @@ class Xcode
     attr_reader :version
     attr_reader :apple_version
     attr_reader :intended_path
+    attr_reader :download_url
 
-    def initialize(semantic_version, intended_path)
+    def initialize(semantic_version, intended_path, download_url)
       @semantic_version = semantic_version
       @intended_path = intended_path
       @apple_version = Xcode::Version.new(@semantic_version).apple
-      @version = XCVersion.available_versions[xcode_index(@apple_version)].strip
+      @download_url = download_url
+      @version = if download_url.empty?
+                   XCVersion.available_versions[xcode_index(@apple_version)].strip
```

Comment:
Let's break this up to make it a little more readable, and avoid some potentially cryptic errors due to the method chaining.

```suggestion
                   version = xcode_index @apple_version
                   listed_version = XCVersion.available_versions[version]
                   listed_version.strip
```